rust_snuba/src/processors/replays.rs

```diff
@@ -0,0 +1,573 @@
+use anyhow::Context;
+use rust_arroyo::backends::kafka::types::KafkaPayload;
+use serde::{Deserialize, Serialize};
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+use uuid::Uuid;
+
+use crate::types::{InsertBatch, KafkaMessageMetadata, RowData};
+
+pub fn process_message(
+    payload: KafkaPayload,
+    metadata: KafkaMessageMetadata,
+) -> anyhow::Result<InsertBatch> {
+    let payload_bytes = payload.payload().context("Expected payload")?;
+
+    let replay_message: ReplayMessage = serde_json::from_slice(payload_bytes)?;
+    let replay_payload = serde_json::from_slice(&replay_message.payload)?;
+
+    let output = match replay_payload {
+        ReplayPayload::ClickEvent(event) => {
+            let replay_rows = event.clicks.into_iter().map(|click| ReplayRow {
+                click_alt: click.alt,
+                click_aria_label: click.aria_label,
+                click_class: click.class,
+                click_id: click.id,
+                click_is_dead: click.is_dead,
+                click_is_rage: click.is_rage,
+                click_node_id: click.node_id,
+                click_role: click.role,
+                click_tag: click.tag,
+                click_testid: click.testid,
+                click_text: click.text,
+                click_title: click.title,
+                event_hash: click.event_hash,
+                offset: metadata.offset,
+                partition: metadata.partition,
+                project_id: replay_message.project_id,
+                replay_id: replay_message.replay_id,
+                retention_days: replay_message.retention_days,
+                timestamp: click.timestamp,
+                ..Default::default()
+            });
+
+            let mut rows: Vec<Vec<u8>> = Vec::with_capacity(replay_rows.len());
+
+            for row in replay_rows {
+                rows.push(serde_json::to_vec(&row)?);
+            }
+
+            rows
+        }
+        ReplayPayload::Event(event) => {
+            let event_hash = match (event.event_hash, event.segment_id) {
+                (None, None) => Uuid::new_v4(),
+                (None, Some(segment_id)) => {
+                    Uuid::from_slice(md5::compute(segment_id.to_string().as_bytes()).as_slice())?
```

Comment:
Its a bit weird that you have an `u16`, and then do an md5 hash of the string representation 🤔 , and then create an uuid from it…
you pretty much expand an `u16` to `16` bytes in a really inefficient way :-D

Reply:
I agree. I'm doing this to satisfy backwards-compatibility requirements with the python processor.  I'd like to remove it once we're happy with the switch.